go/chat/teamchannelsource.go

```diff
@@ -4,225 +4,135 @@ import (
-	c, err := lru.New(100)
-	if err != nil {
-		panic(err)
-	}
-	return &CachingTeamChannelSource{
+func NewTeamChannelSource(g *globals.Context) *TeamChannelSource {
+	return &TeamChannelSource{
 		Contextified: globals.NewContextified(g),
-		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "CachingTeamChannelSource", false),
-		ri:           ri,
-		cache:        c,
-	}
-}
-
-func (c *CachingTeamChannelSource) key(teamID chat1.TLFID, topicType chat1.TopicType) string {
-	return fmt.Sprintf("tid:%v,tt:%v", teamID.String(), int(topicType))
-}
-
-func (c *CachingTeamChannelSource) fetchFromCache(ctx context.Context,
-	teamID chat1.TLFID, topicType chat1.TopicType) (res []chat1.ChannelNameMention, ok bool) {
-	val, ok := c.cache.Get(c.key(teamID, topicType))
-	if !ok {
-		return res, false
-	}
-	var item teamChannelCacheItem
-	if item, ok = val.(teamChannelCacheItem); !ok {
-		return nil, false
+		DebugLabeler: utils.NewDebugLabeler(g.GetLog(), "TeamChannelSource", false),
 	}
-	// c.ChannelsChanged handles cache invalidation on updates.
-	return item.dat, true
 }
 
-func (c *CachingTeamChannelSource) writeToCache(ctx context.Context,
-	teamID chat1.TLFID, topicType chat1.TopicType, names []chat1.ChannelNameMention) {
-	c.cache.Add(c.key(teamID, topicType), teamChannelCacheItem{
-		dat: names,
-	})
+func (c *TeamChannelSource) getTLFConversations(ctx context.Context, uid gregor1.UID,
+	teamID chat1.TLFID, topicType chat1.TopicType) (types.Inbox, error) {
+	inbox, err := c.G().InboxSource.ReadUnverified(ctx, uid, true, /* useLocalData */
+		&chat1.GetInboxQuery{
+			TlfID:            &teamID,
+			TopicType:        &topicType,
+			SummarizeMaxMsgs: false,
+			MemberStatus:     chat1.AllConversationMemberStatuses(),
+		}, nil /* pagination */)
+	return inbox, err
 }
 
-func (c *CachingTeamChannelSource) invalidate(ctx context.Context, teamID chat1.TLFID) {
-	for _, topicType := range chat1.TopicTypeMap {
-		c.cache.Remove(c.key(teamID, topicType))
-	}
-}
+func (c *TeamChannelSource) GetChannelsFull(ctx context.Context, uid gregor1.UID,
+	teamID chat1.TLFID, topicType chat1.TopicType) (res []chat1.ConversationLocal, err error) {
+	defer c.Trace(ctx, func() error { return err }, "GetChannelsFull", res)()
 
-func (c *CachingTeamChannelSource) GetChannelsFull(ctx context.Context, uid gregor1.UID, teamID chat1.TLFID,
-	topicType chat1.TopicType) (res []chat1.ConversationLocal, err error) {
-	var convs []chat1.Conversation
-	tlfRes, err := c.ri().GetTLFConversations(ctx, chat1.GetTLFConversationsArg{
-		TlfID:            teamID,
-		TopicType:        topicType,
-		SummarizeMaxMsgs: false,
-	})
+	inbox, err := c.getTLFConversations(ctx, uid, teamID, topicType)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
-	convs = tlfRes.Conversations
-
-	// Localize the conversations
-	res, _, err = c.G().InboxSource.Localize(ctx, uid, utils.RemoteConvs(convs),
+	convs, _, err := c.G().InboxSource.Localize(ctx, uid, inbox.ConvsUnverified,
```

Comment:
Can just call `Read`

Reply:
fixed

Reply:
reverted, using `Read` makes things more complex since we have to convert the `TlfID` to a name and would have to thread the `MembersType` through for that. (just to convert back to a tlfID when we change the local query to a remote one). think it's simpler as is

Reply:
i started threading it through and i think was hitting errors here (if i'm remember correctly) https://github.com/keybase/client/blob/master/go/chat/inboxsource.go#L27

i also prefer it as is since both `GetChannelsTopicName` and `GetChannelsFull` can use `getTLFConversations` instead of duplicating the query between `Read` and `ReadUnverified` (`GetChannelsTopicName` can't call `Localize` or we get into some unboxing cycle)

Reply:
Sure this is fine.